crates/ruff_cli/src/commands/check.rs

```diff
@@ -56,18 +56,7 @@ pub(crate) fn check(
             }
         }
 
-        match pyproject_config.strategy {
-            PyprojectDiscoveryStrategy::Fixed => {
-                init_cache(&pyproject_config.settings.cli.cache_dir);
-            }
-            PyprojectDiscoveryStrategy::Hierarchical => {
-                for settings in
-                    std::iter::once(&pyproject_config.settings).chain(resolver.settings())
-                {
-                    init_cache(&settings.cli.cache_dir);
-                }
```

Comment:
Is this ok? Do we not use the `cache_dir` of each individual settings?

Reply:
🤷‍♀️ My assumption was that settings in AllSettings are global and not overridable. What's the use case for different caches per directory?

Reply:
Back to the old behavior